ArchitectureFlow/ArchitectureFlow/Common/Promise/Promise.m

```diff
@@ -0,0 +1,16 @@
+#import "Promise.h"
+
+@implementation Promise
+
++ (instancetype)startOnBackgroundWithCompletionBlock:(void(^)(void))blockName {
+    // Abstracting a framework - any third party implementation will do
```

Comment:
No we shouldn't, sorry if the comments didn't make it clear, but we wanted to make it clear we should return some kind of Promise, it can be something be build ourselves, or bolts library or anything else. The point is that we have a receipt